tests/tests_api/test_api.py

```diff
@@ -102,3 +68,79 @@ def test_throw_error_at_source_registration_if_error(
         kirby_app.add_destination(MagicMock(id=1))
     with pytest.raises(ClientError):
         kirby_app.add_destination(MagicMock(id=-1))
+
+
+@patch("requests.session")
+def test_throw_error_at_source_registration_if_error(
+    session_mock, kirby_expected_env, kirby_hidden_env
+):
+    session_mock.return_value.get.side_effect = [
+        MagicMock(
+            status_code=500, json=MagicMock(return_value={"Server Error"})
+        ),
+        MagicMock(
+            status_code=400, json=MagicMock(return_value={"Client Error"})
+        ),
+    ]
+    session_mock.return_value.patch.return_value = MagicMock(
+        status_code=200, json=MagicMock(return_value={})
+    )
+
+    kirby_app = Kirby(kirby_expected_env)
+    with pytest.raises(ServerError):
+        kirby_app.add_source(MagicMock(id=1))
+    with pytest.raises(ClientError):
+        kirby_app.add_source(MagicMock(id=-1))
+
+
+def test_it_raise_error_if_bad_usage_of_testing_mode():
+    kirby_app = Kirby({}, testing=True)
+    with pytest.raises(NotImplementedError):
+        kirby_app.get_topic_id(topic_name="")
+
+
+@patch("requests.session")
+def test_it_get_topic_id(session_mock, kirby_hidden_env):
```

Comment:
what's the point if you're not using the database but mocking this interaction?

Reply:
It is now an integration test, tested with a running instance of kirby-web server